hypothesis-python/tests/common/utils.py

```diff
@@ -137,3 +137,17 @@ def non_covering_examples(database):
     return {
         v for k, vs in database.data.items() if not k.endswith(b".coverage") for v in vs
     }
+
+
+def counts_calls(func):
+    """A decorator that counts how many times a function was called, and
+    stores that value in a ``.calls`` attribute.
```

Comment:
As a sanity check, maybe add `assert not hasattr(func, "calls")` to detect overriding an attribute or double-wrapping?